src/Workspaces/Core/Desktop/Workspace/Storage/PersistentStorageService.cs

```diff
@@ -22,15 +22,12 @@ internal abstract partial class AbstractPersistentStorageService : IPersistentSt
         protected readonly IOptionService OptionService;
         private readonly SolutionSizeTracker _solutionSizeTracker;
 
-        private readonly object _lookupAccessLock;
-        private readonly Dictionary<string, AbstractPersistentStorage> _lookup;
         private readonly bool _testing;
+        private readonly object _lookupAccessLock;
```

Comment:
Should this now be called _primaryStorageLock? It's even more unclear what it's locking now. :smile:

Reply:
sure I can. it is locking updating primaryStorage data. didn't move inside of PrimaryStorageInfo since I saw this - http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Workspace/Host/PersistentStorage/AbstractPersistentStorage.cs,70